src/main/java/duke/DukeIO.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class DukeIO {
+
+    private static final String DUKEDATADIR = "data";
+    private static final String DUKEDATA = "data/dukeData.txt";
+
+    // create file and directory if they are not created yet
+    public static void initialize() throws IOException {
+        File dataDirectory = new File(DUKEDATADIR);
+        if (!dataDirectory.exists()){
+            dataDirectory.mkdirs();
+        }
+        File dukeData = new File(DUKEDATA);
+        if(!dukeData.exists()){
+            dukeData.createNewFile();
+        }
+    }
+
+    public static void writeDukeData() throws IOException {
```

Comment:
This method is rather long, with quite a number of if-else statements; perhaps you could consider trying to break this method down into further methods, or further simplifying it.